drivers: flash_stm32_xspi: fix DT accessor for flash size

tpambor opened the pull request:
This PR is based on #76735 which updated STM flash drivers for QSPI and OSPI but not for XSPI. #87466 then fixed a regression for boards that use QSPI and OSPI but at the same time broke boards with XSPI. This PR aligns XSPI again with QSPI and OSPI and fixes the compile errors currently observed on main for the sample `application_development/code_relocation_nocopy`:

```
/opt/toolchains/zephyr-sdk-0.17.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: invalid length for memory region EXTFLASH
```

erwango commented:
Thanks @tpambor for pushing this fix.

However, merging it will get dtc generate warnings (see https://github.com/zephyrproject-rtos/zephyr/issues/88404). So the choice is:
- fix this particular use case and get warnings
- revert https://github.com/zephyrproject-rtos/zephyr/pull/76735 and https://github.com/zephyrproject-rtos/zephyr/pull/87466 (but the issue it was supposed to fix will re-surface)
- revert https://github.com/zephyrproject-rtos/zephyr/pull/68274 (out of tree impacts to be considered)

Considering this is not a decision that should be taken lightly, I propose to merge this PR and add a temporary rule to avoid the warnings the time we settle on the direction to be followed.

tpambor committed:
drivers: flash_stm32_xspi: fix DT accessor for flash size

The flash size is the second part (size) of the first reg value, not the
first part (address) of a nonexistent second reg value.

Based-on-patch-by: Armin Brauns <armin.brauns@embedded-solutions.at>
Signed-off-by: Tim Pambor <tim.pambor@codewrights.de>

erwango commented:
About:
> add a temporary rule to avoid the warnings the time we settle on the direction to be followed.

Can't find a way to avoid the warning globally on the whole family, so this has to be done per board.. I'd prefer to live with the warnings for the time being.

tpambor commented:
I agree to merge this first as other PRs for these boards are blocked by the pipeline failures.
